rns_shell/compositor/layers/Layer.cpp

```diff
@@ -214,7 +214,11 @@ bool Layer::needsPainting(PaintContext& context) {
       return false;
     }
 
+#if USE(RNS_SHELL_PARTIAL_UPDATES)
+    if(!context.supportPartialUpdate && (context.damageRect.size() == 0)) // No damage rect set, so need to paint the layer if partial update disabled
```

Comment:
We don't need the second condition. We don't add any damage rect when supportPartialUpdate is false